pkg/webhook/v1alpha3/pod/inject_webhook.go

```diff
@@ -199,3 +201,110 @@ func (s *sidecarInjector) getMetricsCollectorImage(cKind common.CollectorKind) (
 		return "", errors.New("Failed to find metrics collector configuration in configmap " + consts.KatibConfigMapName)
 	}
 }
+
+func getMetricsCollectorArgs(trialName, metricName string, mc trialsv1alpha3.MetricsCollectorSpec) []string {
+	args := []string{"-t", trialName, "-m", katibmanagerv1alpha3.GetManagerAddr(), "-mn", metricName}
+	if mountFile := getMountFile(mc); mountFile != "" {
+		args = append(args, "-f", mountFile)
+	}
+	return args
+}
+
+func getMountFile(mc trialsv1alpha3.MetricsCollectorSpec) string {
+	if mc.Collector.Kind == common.StdOutCollector {
+		return common.DefaultFilePath
+	} else if mc.Collector.Kind == common.FileCollector {
+		return mc.Source.FileSystemPath.Path
+	} else {
+		return ""
+	}
+}
+
+func wrapWorkerContainer(pod *v1.Pod, jobKind, metricsFile string, mc trialsv1alpha3.MetricsCollectorSpec) {
+	if mc.Collector.Kind != common.StdOutCollector {
```

Comment:
What does the function do

Reply:
this function aims to redirect stdout to a file so that both StdoutCollector and FileCollector can share same image to parse file to get metrics.
I will refactor this function to handle container.command is nil case, which will parse image entrypoint just like how Tekton project handle the case